components/datetime/src/scaffold/into_option.rs

```diff
@@ -0,0 +1,153 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::neo_skeleton::*;
+use icu_calendar::{types::*, AnyCalendarKind, Date, Iso};
+use icu_timezone::{TimeZoneBcp47Id, UtcOffset, ZoneVariant};
+
+/// Converts Self to an `Option<T>`, either `Some(T)` if able or `None`
+pub trait IntoOption<T> {
+    /// Return `self` as an `Option<T>`
+    fn into_option(self) -> Option<T>;
+}
+
+impl<T> IntoOption<T> for Option<T> {
+    #[inline]
+    fn into_option(self) -> Option<T> {
+        self
+    }
+}
+
+impl<T> IntoOption<T> for () {
+    #[inline]
+    fn into_option(self) -> Option<T> {
+        None
+    }
+}
+
+impl IntoOption<YearInfo> for YearInfo {
```

Comment:
can these be macro-generated?

Reply:
Maybe, but I'm moving them to another crate in the next PR